core/templates/domain/exploration/InteractionObjectFactory.ts

```diff
@@ -92,14 +92,14 @@ export class Interaction {
   customizationArgs: InteractionCustomizationArgs;
   defaultOutcome: Outcome | null;
   hints: Hint[];
-  id: string | null;
+  id: string;
```

Comment:
Are you sure this can't be null? If you create a new exploration and don't make any changes then the interaction ID for the first state would presumably be null since it has not been specified yet.

Reply:
#14464 This issue require me to delete this null. If this string can be null, do we really need to remove null as an option?

> Are you sure this can't be null? If you create a new exploration and don't make any changes then the interaction ID for the first state would presumably be null since it has not been specified yet.

Reply:
I think the issue author is partially correct. There are some places where interactionID can be null and some where it cannot be. In the latter places, null should not be allowed as an option, though it's fine to have it as an option in the former places.

Reply:
ok, so I should keep this null and find the correct place to delete.